lib/Dialect/ESI/runtime/python/esiaccel/cosim/verilator.py

```diff
@@ -14,14 +14,10 @@ class Verilator(Simulator):
 
   DefaultDriver = CosimCollateralDir / "driver.cpp"
 
-  def __init__(self, sources: SourceFiles, run_dir: Path, debug: bool):
-    super().__init__(sources, run_dir, debug)
-
+  def compile_commands(self) -> List[List[str]]:
     self.verilator = "verilator"

```

Comment:
Why'd you move this? I like to set up everything in constructors so that if they happen to get used in other places they can be. Set up shouldn't be a side-effect of a function not called by the constructor. If you're gonna move it, it should be a local variable.

Reply:
To avoid complicating the constructor (i.e. duplicating the `Simulator` constructor in `Verilator`). But i get your point - I'll reconsider!

Reply:
You *could* pass all of the args by `kwargs` but that fucks with code completion so I prefer the solution you've done here.